src/common/reconciler.go

```diff
@@ -229,58 +202,83 @@ func (r *ServiceReconciler) updateService(service *opslevel.Service, registratio
 		log.Warn().Msgf("[%s] unexpected happened: service passed to be updated is nil", registration.Name)
 		return
 	}
-	updateServiceInput := opslevel.ServiceUpdateInput{
-		Id:          &service.Id,
-		Product:     opslevel.RefOf[string](registration.Product),
-		Description: opslevel.RefOf[string](registration.Description),
-		Language:    opslevel.RefOf[string](registration.Language),
-		Framework:   opslevel.RefOf[string](registration.Framework),
+
+	// updateServiceInput contains the changes needed to reconcile the service
+	updateServiceInput := opslevel.ServiceUpdateInput{Id: &service.Id}
+	// for each field - check if the value exists in the registration AND if the value has changed compared to what is currently set
+	if registration.Description != "" && registration.Description != service.Description {
+		updateServiceInput.Description = opslevel.RefOf(registration.Description)
+	}
+	if registration.Framework != "" && registration.Framework != service.Framework {
+		updateServiceInput.Framework = opslevel.RefOf(registration.Framework)
+	}
+	if registration.Language != "" && registration.Language != service.Language {
+		updateServiceInput.Language = opslevel.RefOf(registration.Language)
+	}
+	if registration.Lifecycle != "" && registration.Lifecycle != service.Lifecycle.Alias {
```

Comment:
this is an unnecessary logic gate as the original code already checks for this AND the `cmp.Diff` will give us the 2nd part.